plugins/modules/redfish_firmware_rollback.py

```diff
@@ -0,0 +1,360 @@
+    "TargetSettingsURI": null
+  }]
+error_info:
+  type: dict
+  description: Details of the HTTP error.
+  returned: on http error
+  sample: {
+    "error": {
+      "@Message.ExtendedInfo": [{
+        "Message": "InstanceID value provided for the update operation is invalid",
+        "MessageArgs": [],
+        "MessageArgs@odata.count": 0,
+        "MessageId": "IDRAC.2.8.SUP024",
+        "RelatedProperties": [],
+        "RelatedProperties@odata.count": 0,
+        "Resolution": "Enumerate inventory, copy the InstanceID value and provide that value for the update operation.",
+        "Severity": "Warning"
+      }],
+      "code": "Base.1.12.GeneralError",
+      "message": "A general error has occurred. See ExtendedInfo for more information"
+    }
+  }
+"""
+
+
+import json
+import re
+import time
+import sre_constants
+from ssl import SSLError
+from ansible_collections.dellemc.openmanage.plugins.module_utils.redfish import Redfish, redfish_auth_params, \
+    SESSION_RESOURCE_COLLECTION
+from ansible_collections.dellemc.openmanage.plugins.module_utils.utils import wait_for_redfish_reboot_job, \
+    wait_for_redfish_job_complete, strip_substr_dict, MANAGER_JOB_ID_URI, RESET_UNTRACK, MANAGERS_URI, RESET_SUCCESS
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import ConnectionError, SSLValidationError
+from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
+
+
+UPDATE_SERVICE = "UpdateService"
+SYSTEM_RESOURCE_ID = "System.Embedded.1"
+NO_COMPONENTS = "There were no firmware components to rollback."
+CHANGES_FOUND = "Changes found to be applied."
+NO_CHANGES_FOUND = "No changes found to be applied."
+NOT_SUPPORTED = "The target firmware version does not support the firmware rollback."
+COMPLETED_ERROR = "The job for firmware rollback has been completed with error(s)."
+SCHEDULED_ERROR = "The job for firmware rollback has been scheduled with error(s)."
+ROLLBACK_SUCCESS = "Successfully completed the job for firmware rollback."
+ROLLBACK_SCHEDULED = "Successfully scheduled the job for firmware rollback."
+ROLLBACK_FAILED = "Failed to complete the job for firmware rollback."
+REBOOT_FAIL = "Failed to reboot the server."
+NEGATIVE_TIMEOUT_MESSAGE = "The parameter reboot_timeout value cannot be negative or zero."
+JOB_WAIT_MSG = "Task excited after waiting for {0} seconds. Check console for firmware rollback status."
+REBOOT_COMP = ["Integrated Dell Remote Access Controller"]
+
+
+def get_rollback_preview_target(redfish_obj, module):
+    action_resp = redfish_obj.invoke_request("GET", "{0}{1}".format(redfish_obj.root_uri, UPDATE_SERVICE))
+    action_attr = action_resp.json_data["Actions"]
+    update_uri = None
+    if "#UpdateService.SimpleUpdate" in action_attr:
+        update_service = action_attr.get("#UpdateService.SimpleUpdate")
+        if 'target' in update_service:
+            update_uri = update_service.get('target')
+        else:
```

Comment:
no need of else here, even if you remove it'll go to next line and fail_json will trigger.

Reply:
okay got it, but could have handle in negative sense in 180, so we can avoid one else statement.

like below to avoid else statement. just a suggestion

if` 'target' not in update_service:
           module.fail_json(msg=NOT_SUPPORTED)
update_uri = update_service.get('target')

Reply:
code will be updated with this suggestion

Reply:
Updated